src/Serilog.Enrichers.CorrelationId/Middlewares/CorrelationIdHeaderSupplierMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+#if NETCORE
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Serilog.Enrichers.CorrelationId.Middlewares
+{
+    public class CorrelationIdHeaderSupplierMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly string _headerKey;
+
+        public CorrelationIdHeaderSupplierMiddleware(RequestDelegate next, string headerKey = "x-correlation-id")
+        {
+            _next = next;
+            _headerKey = headerKey;
+        }
+
+        public Task Invoke(HttpContext httpContext)
+        {
+            string correlationId = Guid.NewGuid().ToString();
```

Comment:
This should be null until the below if() evaluates, as currently, it could be overwritten, making this allocation pointless

Reply:
fixed in d667c8e